main.go

```diff
@@ -32,12 +36,47 @@ func init() {
 		xlog.NewHandler(cfg.loggerConfig),
 		xaccess.NewHandler(),
 	)
+	sdnotify.Ready()
+	watchdog(uno)
+}
+
+
+func watchdog(uno *unoconv) {
+	var watchdog_usec time.Duration
+
+	if value, ok := os.LookupEnv("WATCHDOG_USEC"); ok {
+		if v, err := strconv.Atoi(value); err == nil {
+			watchdog_usec = time.Duration(v) * time.Microsecond
+		} else {
+			return
+		}
+	} else {
+		return
+	}
+
+	go func(uno *unoconv, watchdog_usec time.Duration) {
+		for {
+			time.Sleep(watchdog_usec / 2)
```

Comment:
Is it necessary to call the watchdog every x microseconds? 
Seems to me to be quite frequent.

Reply:
WATCHDOG_USEC is configured in the systemd unit in seconds with 
https://www.freedesktop.org/software/systemd/man/systemd.service.html#WatchdogSec=